src/main/java/com/exasol/repository/ReleaseChangesLetter.java

```diff
@@ -0,0 +1,164 @@
+    private final String fileName;
+    private String header = null;
+    private String body = null;
+    private String versionNumber = null;
+    private LocalDate releaseDate = null;
+    private final List<Integer> ticketNumbers = new ArrayList<>();
+
+    /**
+     * Create a new {@link ReleaseChangesLetter} from a string.
+     * 
+     * @param fileName name of the changes file
+     * @param content file's content as a string
+     */
+    public ReleaseChangesLetter(final String fileName, final String content) {
+        this.fileName = fileName;
+        if (content != null && content.length() > 1) {
+            parseContent(content);
+        }
+    }
+
+    /**
+     * Get a name of the changes file.
+     * 
+     * @return name of the changes file
+     */
+    public String getFileName() {
+        return this.fileName;
+    }
+
+    /**
+     * Get a version number.
+     * 
+     * @return version number as a string
+     */
+    public Optional<String> getVersionNumber() {
+        return this.versionNumber == null ? Optional.empty() : Optional.of(this.versionNumber);
+    }
+
+    /**
+     * Get a release date.
+     * 
+     * @return release date
+     */
+    public Optional<LocalDate> getReleaseDate() {
+        return this.releaseDate == null ? Optional.empty() : Optional.of(this.releaseDate);
+    }
+
+    /**
+     * Get a release letter header.
+     * 
+     * @return header as a string
+     */
+    public Optional<String> getHeader() {
+        return this.header == null ? Optional.empty() : Optional.of(this.header);
+    }
+
+    /**
+     * Get a release letter body.
+     * 
+     * @return body as a atring
+     */
+    public Optional<String> getBody() {
+        return this.body == null ? Optional.empty() : Optional.of(this.body);
+    }
+
+    /**
+     * Get mentioned tickets' numbers.
+     * 
+     * @return list of ticket numbers
+     */
+    public List<Integer> getTicketNumbers() {
+        return this.ticketNumbers;
+    }
+
+    private void parseContent(final String content) {
+        final List<String> contentParts = divideContent(content);
+        if (contentParts.size() == 2) {
+            parseHeaders(contentParts.get(0));
+            parseBody(contentParts.get(1));
+        }
+    }
+
+    private List<String> divideContent(final String content) {
+        final int divideIndex = content.indexOf("##");
+        if (divideIndex != -1) {
+            final String headersPart = content.substring(0, divideIndex);
+            final String bodyPart = content.substring(divideIndex);
+            return List.of(headersPart, bodyPart);
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    private void parseHeaders(final String content) {
+        parseVersionNumber(content);
+        parseReleaseDate(content);
+        parseHeader(content);
+    }
+
+    private void parseBody(final String content) {
+        this.body = content;
```

Comment:
Do not assign class fields as a side effect. This creates illegal state in a class which forces users to call methods in correct order.

Reply:
I recommend splitting of the parser and turning the `ReleaseLetter` class into a pure parameter object. Add a builder to the class and let the parser construct it using the builder.

Reply:
I was thinking about that!

Reply:
Saw the implementation. If you additionally turn the parser into a dynamic class, I am happy.

Reply:
done